DOCS-7600-applied template (#209)
mulesoft/docs-api-manager: modules/ROOT/pages/apply-configure-json-threat-task.adoc

>s|Policy Name|JSON Threat Protection
>s|Summary|Protects against malicious JSON in API requests
>s|Category|Security
>s|Available from Mule version|3.8.0
>s|Returned Status Codes|400 - Bad Request
|===

> JSON requests are susceptible to attacks characterized by unusual inflation of elements and nesting levels. Attackers use recursive techniques to consume memory resources. Dramatic swings in the size of the application data often signal a security problem. The JSON Threat Protection policy helps protect your applications from such intrusions.

Reviewer: "JSON requests are susceptible to attacks" The problem is not on the requests themselves but on the applications processing those requests. Something more accurate would be: Application consuming JSON based request are susceptible to attacks… Or something in those lines.

Author: Done! Thanks!

> If Mule runtime engine (Mule) fails to detect an attack, you must monitor and design your services architecture with layers of protection in addition to these policies.

Reviewer: "If Mule runtime engine (Mule) fails to detect an attack": This statement seems to imply that there is in place a complex system that comes out of the box (or can be enabled somehow), that monitors for attacks in general, which is not correct. No such system exists today or will exist in the near future. Perhaps you may say “There are other possible attack vectors. Consider designing your services architecture with other layers of protection in addition to this one.”

Author: Done! Thanks!

== How This Policy Works

//anything to add here -- nahuel

//which policies--nahuel

== Configuring Policy Parameters

When you apply the policy to your API from the UI, the following parameters are displayed:

[width="100%", cols="2,6,1,1"]

> |===

Reviewer: Default values on the table are incorrect. The default is not "0" but "-1".

Author: Done! Thanks!

>s|Policy Name|JSON Threat Protection
>s|Summary|Protects against malicious JSON in API requests
>s|Category|Security

> >s|Available from Mule version|3.8.0

Reviewer 2: Do not use "from" in this context, to mean "starting with," because "from" has other similar meanings.
Replace with "Available in Product Name + x.x.x + and later"

Author: That is how it is in the file that I see. I don't know why it did not update it for you.

>s|Returned Status Codes|400 - Bad Request
|===

Applications processing JSON requests are susceptible to attacks characterized by unusual inflation of elements and nesting levels. Attackers use recursive techniques to consume memory resources. Dramatic swings in the size of the application data often signal a security problem. The JSON Threat Protection policy helps protect your applications from such intrusions.

> If any possible attacks to your Anypoint Platform setup go undetected, consider designing your services architecture with layers of protection in addition to these policies.

Reviewer 2: ```suggestion
If you find that attacks on your Anypoint Platform setup are difficult to detect, design your services architecture with layers of protection in addition to JSON Threat Protection.
```

Author: Done! Thanks!

> == How This Policy Works

Reviewer 2: WHICH policy? Is this about JSON or about "layers of protection in addition to"?
Name the policy in the title, so that someone scanning will know to what "this" refers.

Author: Removing this section as there is nothing to document in that part.

//anything to add here -- nahuel

//which policies--nahuel

== Configuring Policy Parameters

> When you apply the policy to your API from the UI, the following parameters are displayed:

Reviewer 2: Again, which policy? Because your topic introduces both JSON and "additional policies," this is unclear.
Maybe you should not mention them in the beginning?

Author: Done! It is about the JSON Threat Protection policy

[width="100%", cols="2,6,1,1"]
|===

> s|Field s|Description s|Default s|Required

Reviewer 2: What is up with the "s"s?

Author: Not sure myself. But the output looks fine.

> |Maximum Container Depth|Specifies the maximum allowed nested depth. JSON allows you to nest the containers (object and array) in any order to any depth ^|-1 ^|false

Reviewer 2: ```suggestion
|Maximum Container Depth|Specifies the maximum nested depth. JSON allows you to nest the containers (object and array) in any order to any depth |-1 |false
```
The default is negative 1?

Author: Yes! Done! Thanks!

> |Maximum String Value Length|Specifies the maximum length allowed for a string value ^|-1 ^|false

Reviewer 2: ```suggestion
|Maximum String Value Length|Specifies the maximum length of a string value ^|-1 ^|false
```
ditto GLOBAL about the ^ symbols ("s")

Author: Done! Thanks!

|Maximum Object Entry Name Length|Specifies the maximum string length allowed for an object's entry name ^|-1 ^|false
|Maximum Object Entry Count|Specifies the maximum number of entries allowed in an object ^|-1 ^|false

> |Maximum Array Element Count|Specifies the maximum number of elements allowed in an array ^|-1 ^|false

Reviewer 2: ditto all previous

Author: Done! Thanks!
